include/gpioMonitor.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <sdbusplus/asio/connection.hpp>
+
+namespace vpd
+{
+/**
+ * @brief class for GPIO event handling.
+ *
+ * Responsible for detecting events and handling them. It continuously
+ * monitors the presence of the FRU. If any attachment or detachment is
+ * detected, it enables or disables the FRU's output GPIO and binds or
+ * unbinds the driver accordingly.
+ */
+class GpioEventHandler
+{
+  public:
+    GpioEventHandler() = delete;
+    ~GpioEventHandler() = delete;
+    GpioEventHandler(const GpioEventHandler&) = delete;
+    GpioEventHandler& operator=(const GpioEventHandler&) = delete;
+    GpioEventHandler(GpioEventHandler&&) = delete;
+    GpioEventHandler& operator=(GpioEventHandler&&) = delete;
+
+    /**
+     * @brief Constructor
+     *
+     * @param[in] i_presPin - GPIO presence pin
+     * @param[in] i_presevalue - holds the value of presence pin
+     * @param[in] i_outPin - output pin
+     * @param[in] i_outValue - value of the output pin
+     * @param[in] i_inventoryPsth - inventory path of the FRU.
+     * @param[in] i_ioCon - pointer to the io context object
+     */
+    GpioEventHandler(std::string& i_presencePin, bool& i_presenceValue,
+                     std::string& i_outputPin, bool& i_outputValue,
+                     std::string& i_inventoryPath,
+                     std::shared_ptr<boost::asio::io_context>& i_ioContext) :
+        m_presencePin(i_presencePin),
+        m_presenceValue(i_presenceValue), m_outputPin(i_outputPin),
+        m_outputValue(i_outputValue), m_inventoryPath(i_inventoryPath)
+    {
+        setEventHandlerForGpioPresence(i_ioContext);
+    }
+
+  private:
+    /** @brief API to read the GPIO presence pin value.
+     *
+     *  @returns GPIO presence pin value.
+     */
+    bool getPresencePinValue();
+
+    /** @brief API to toggle the output gpio as per the presence
```

Comment:
Same here and brief should be wrapped in a line.

Reply:
Done